php-frontend/src/main/java/org/sonar/php/parser/TreeFactory.java

```diff
@@ -0,0 +1,1100 @@
+  }
+
+  public CompoundVariableTree compoundVariable(InternalSyntaxToken openDollarCurly, ExpressionTree expression, InternalSyntaxToken closeDollarCurly) {
+    return new CompoundVariableTreeImpl(openDollarCurly, expression, closeDollarCurly);
+  }
+
+  public ArrayAccessTree dimensionalOffset(InternalSyntaxToken openCurly, Optional<ExpressionTree> expression, InternalSyntaxToken closeCurly) {
+    if (expression.isPresent()) {
+      return new ArrayAccessTreeImpl(openCurly, expression.get(), closeCurly);
+    }
+    return new ArrayAccessTreeImpl(openCurly, closeCurly);
+  }
+
+  public ExpressionTree variableWithoutObjects(Optional<List<InternalSyntaxToken>> dollars, VariableTree compoundVariable, Optional<List<ArrayAccessTree>> offsets) {
+    ExpressionTree result = compoundVariable;
+    for (ExpressionTree partialArrayAccess : optionalList(offsets)) {
+      result = ((ArrayAccessTreeImpl) partialArrayAccess).complete(result);
+    }
+
+    if (dollars.isPresent()) {
+      result = new VariableVariableTreeImpl(dollars.get(), result);
+    }
+
+    return result;
+  }
+
+  public ArrayAccessTree alternativeDimensionalOffset(InternalSyntaxToken openBrace, Optional<ExpressionTree> offset, InternalSyntaxToken closeBrace) {
+    if (offset.isPresent()) {
+      return new ArrayAccessTreeImpl(openBrace, offset.get(), closeBrace);
+    }
+    return new ArrayAccessTreeImpl(openBrace, closeBrace);
+  }
+
+  public IdentifierTree newStaticIdentifier(InternalSyntaxToken staticToken) {
+    return new IdentifierTreeImpl(staticToken);
+  }
+
+  public ReferenceVariableTree referenceVariable(InternalSyntaxToken ampersand, ExpressionTree variable) {
+    return new ReferenceVariableTreeImpl(ampersand, variable);
+  }
+
+  public SpreadArgumentTree spreadArgument(InternalSyntaxToken ellipsis, ExpressionTree expression) {
+    return new SpreadArgumentTreeImpl(ellipsis, expression);
+  }
+
+  public FunctionCallTree functionCallParameterList(InternalSyntaxToken openParenthesis, Optional<Tuple<ExpressionTree, Optional<List<Tuple<InternalSyntaxToken, ExpressionTree>>>>> arguments, InternalSyntaxToken closeParenthesis) {
+    SeparatedList<ExpressionTree> list;
+    if (arguments.isPresent()) {
+      list = separatedList(arguments.get().first(), arguments.get().second());
+    } else {
+      list = SeparatedList.empty();
+    }
+
+    return new FunctionCallTreeImpl(openParenthesis, list, closeParenthesis);
+  }
+
+  public MemberAccessTree classMemberAccess(InternalSyntaxToken token, Tree member) {
+    return new MemberAccessTreeImpl(Kind.CLASS_MEMBER_ACCESS, token, member);
+  }
+
+  public ExpressionTree objectDimensionalList(ExpressionTree variableName, Optional<List<ArrayAccessTree>> dimensionalOffsets) {
+    ExpressionTree result = variableName;
+
+    for(ArrayAccessTree arrayAccess : optionalList(dimensionalOffsets)) {
+       result = ((ArrayAccessTreeImpl) arrayAccess).complete(result);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make this line start at column 7. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AIndentationCheck)